src/freenas/usr/local/libexec/nas/generate_smb4_conf.py

```diff
@@ -1772,8 +1749,21 @@ def main():
     generate_smb4_system_shares(client, smb4_shares)
     generate_smb4_shares(client, smb4_shares, shares)
 
-    if role == 'dc' and not client.call('notifier.samba4', 'domain_provisioned'):
-        provision_smb4(client)
+    if role == 'dc':
+        """
+        Migrate from legacy sentinel file to ZFS User Property.
+        If the domain has already been provisioned, domaincontroller.provision
+        will return false. In this case we don't need to set a new password or adjust
+        min password requirements. Provisioning raises an exception in case of failure.
+        """
+        if client.call('notifier.samba4', 'domain_provisioned'):
+            client.call('domaincontroller.set_provisioned', True)
+            client.call('notifier.samba4', 'domain_sentinel_file_remove')
+
+        if client.call('domaincontroller.provision'):
```

Comment:
This may raise an exception (CallError), do we want to break smb4.conf gen altogether if it does?